"error  'Set' is not defined  no-undef" in ecmaVersion 2017

**Tell us about your environment**

* **ESLint Version:**
└─┬ gulp-eslint@3.0.1
  └── eslint@3.12.2

* **Node Version:**
v4.2.6

* **npm Version:**
npm: '3.5.2'

**What parser (default, Babel-ESLint, etc.) are you using?**

gulp-eslint

**Please show your full configuration:**

<!-- Paste your configuration below: -->
```

env:
  browser: true                       # Enable Browser global variables

parserOptions:
  ecmaVersion: 2017                   # Validate ECMAScript 2016 features
  sourceType: module                  # Allow ECMAScript modules

globals:
  PIXI: false
  Phaser: false

extends: 'eslint:recommended'

rules:
  indent: [2, 2]                      # Indent code with 2 spaces
  quotes: [2, single]                 # Use single quotes for strings
  linebreak-style: [2, unix]          # Favor Unix-style line endings
  semi: [2, always]                   # End lines with semicolons
```

**What did you do? Please include the actual source code causing the issue.**

<!-- Paste the source code below: -->
```js
  static getLockedBirds(game){
    var allBirdIds=new Set();
    for(let i=0;i<=game.animationInfo.maxBirdFrame;i++){
      allBirdIds.add(i);
    }

    var unlockedBirds = new Set(DataAccess.getConfig('unlockedBirdSprites'));
    var lockedBirds = [...allBirdIds].filter(x => !unlockedBirds.has(x)); //find all bird ids that are not in the unlocked set but ARE in the allBird set

    return lockedBirds;
  }
```

**What did you expect to happen?**
Set is a ECMA 2015 object. By using ECMA 8/2017 I expected the parser to recognize this. If I add `es6: true` to the environment configuration, no error is thrown

**What actually happened? Please include the actual, raw output from ESLint.**
```bash
/home/james/Documents/birdu/src/objects/Helpers/DataAccess.js
  31:24  error  'Set' is not defined  no-undef
  36:29  error  'Set' is not defined  no-undef
```

--- a/README.md
+++ b/README.md
@@ -205,7 +205,7 @@ Yes, ESLint natively supports parsing JSX syntax (this must be enabled in [confi
 
 ### What about ECMAScript 6 support?
 
-ESLint has full support for ECMAScript 6. By default, this support is off. You can enable ECMAScript 6 support through [configuration](http://eslint.org/docs/user-guide/configuring).
+ESLint has full support for ECMAScript 6. By default, this support is off. You can enable ECMAScript 6 syntax and global variables through [configuration](http://eslint.org/docs/user-guide/configuring).
 
 ### What about experimental features?
 

--- a/docs/user-guide/configuring.md
+++ b/docs/user-guide/configuring.md
@@ -18,7 +18,10 @@ All of these options give you fine-grained control over how ESLint treats your c
 ESLint allows you to specify the JavaScript language options you want to support. By default, ESLint expects ECMAScript 5 syntax. You can override that setting to enable support for other ECMAScript versions as well as JSX by using parser options.
 
 Please note that supporting JSX syntax is not the same as supporting React. React applies specific semantics to JSX syntax that ESLint doesn't recognize. We recommend using [eslint-plugin-react](https://github.com/yannickcr/eslint-plugin-react) if you are using React and want React semantics.
-
+By the same token, supporting ES6 syntax is not the same as supporting new ES6 globals (e.g., new types such as
+`Set`).
+For ES6 syntax, use `{ "parserOptions": { "ecmaVersion": 6 } }`; for new ES6 global variables, use `{ "env":
+{ "es6": true } }` (this setting enables ES6 syntax automatically).
 Parser options are set in your `.eslintrc.*` file by using the `parserOptions` property. The available options are:
 
 * `ecmaVersion` - set to 3, 5 (default), 6, 7, or 8 to specify the version of ECMAScript syntax you want to use. You can also set to 2015 (same as 6), 2016 (same as 7), or 2017 (same as 8) to use the year-based naming.

